### PR TITLE
ACC-120 Use acceptance Apache for ssl offloading

### PR DESCRIPTION
**Pull request by madhifallah**
Added templates : 
instance-chat-standalone-cmis-oo.include.template
instance-chat-standalone-oo.include.template
instance-ws-cmis-oo.include.template
instance-ws-oo.include.template
instance-cmis-oo.include.template 
instance.include-oo.template

### CODE REVIEW

File: etc/apache2/includes/instance-chat-standalone-oo.include.template [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    ProxyPassMatch    "^\/(v[\d]+\.[\d]+\.[\d]+[\.|-][\d]+\/.*)$"    http://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1 acquire=1000 retry=30
+    ProxyPassMatch	  "^\/OfficeWeb\/.*)$"							 http://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1 acquire=1000 retry=30
+
+    # Add header for X-Forwarded-For / X-Forwarded-Host / X-Forwarded-Server
+    ProxyAddHeaders On
+    # Add header X-Forwarded-Proto to enforce https on the underlying webserver
+    RequestHeader set X-Forwarded-Proto "https"
+
+    # Proxify of websocket url pattern
+    ProxyPassMatch    (.*)(\/websocket)$                            ws://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1$2
```
Comment (vsellier) [inline etc/apache2/includes/instance-chat-standalone-oo.include.template]: should be on the ```IfModule mod_proxy_wstunnel.c`` section + specific ws configuration

File: etc/apache2/includes/instance-oo.include.template [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    ProxyPass               /images/Arrow.gif    !
+    ProxyPass               /images/BgBlock.jpg  !
+    ProxyPass               /images/Header.png   !
+    ProxyPass               /images/Footer.png   !
+    ProxyPass               /images/Logo.png     !
+
+    ProxyPass               /elasticsearch       http://localhost:${DEPLOYMENT_ES_HTTP_PORT}
+    ProxyPassReverse        /elasticsearch       http://localhost:${DEPLOYMENT_ES_HTTP_PORT}
+
+	# Forward request to OnlyOffice when the url start with the version scheme (see in side the Docker container for it)
```
Comment (vsellier) [inline etc/apache2/includes/instance-oo.include.template]: please check the indentation of this line and the following

File: etc/apache2/includes/instance-oo.include.template [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	ProxyPassMatch    "^\/(v[\d]+\.[\d]+\.[\d]+[\.|-][\d]+\/.*)$"    http://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1 acquire=1000 retry=30
+	ProxyPassMatch	  "^\/OfficeWeb\/.*)$"							 http://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1 acquire=1000 retry=30
+
+	# Add header for X-Forwarded-For / X-Forwarded-Host / X-Forwarded-Server
+	ProxyAddHeaders On
+	# Add header X-Forwarded-Proto to enforce https on the underlying webserver
+	RequestHeader set X-Forwarded-Proto "https"
+
+	# Proxify of websocket url pattern
+	ProxyPassMatch    (.*)(\/websocket)$                            ws://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1$2
```
Comment (vsellier) [inline etc/apache2/includes/instance-oo.include.template]: you should specify the specific ws configuration

File: etc/apache2/includes/instance-cmis-oo.include.template [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    ProxyPassMatch    "^\/(v[\d]+\.[\d]+\.[\d]+[\.|-][\d]+\/.*)$"    http://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1 acquire=1000 retry=30
+    ProxyPassMatch	  "^\/OfficeWeb\/.*)$"							 http://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1 acquire=1000 retry=30
+
+    # Add header for X-Forwarded-For / X-Forwarded-Host / X-Forwarded-Server
+    ProxyAddHeaders On
+    # Add header X-Forwarded-Proto to enforce https on the underlying webserver
+    RequestHeader set X-Forwarded-Proto "https"
+
+    # Proxify of websocket url pattern
+    ProxyPassMatch    (.*)(\/websocket)$                            ws://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1$2
```
Comment (vsellier) [inline etc/apache2/includes/instance-cmis-oo.include.template]: you should specify the specific ws configuration

File: etc/apache2/includes/instance-ws-cmis-oo.include.template [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    ProxyPassMatch    "^\/(v[\d]+\.[\d]+\.[\d]+[\.|-][\d]+\/.*)$"    http://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1 acquire=1000 retry=30
+    ProxyPassMatch	  "^\/OfficeWeb\/.*)$"							 http://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1 acquire=1000 retry=30
+
+    # Add header for X-Forwarded-For / X-Forwarded-Host / X-Forwarded-Server
+    ProxyAddHeaders On
+    # Add header X-Forwarded-Proto to enforce https on the underlying webserver
+    RequestHeader set X-Forwarded-Proto "https"
+
+    # Proxify of websocket url pattern
+    ProxyPassMatch    (.*)(\/websocket)$                            ws://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1$2
```
Comment (vsellier) [inline etc/apache2/includes/instance-ws-cmis-oo.include.template]: * this should be on the ``<IfModule mod_proxy_wstunnel.c>`` section
* you should add the specific ws configuration

File: etc/apache2/includes/instance-ws-oo.include.template [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	ProxyPassMatch    "^\/(v[\d]+\.[\d]+\.[\d]+[\.|-][\d]+\/.*)$"    http://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1 acquire=1000 retry=30
+	ProxyPassMatch	  "^\/OfficeWeb\/.*)$"							 http://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1 acquire=1000 retry=30
+
+	# Add header for X-Forwarded-For / X-Forwarded-Host / X-Forwarded-Server
+	ProxyAddHeaders On
+	# Add header X-Forwarded-Proto to enforce https on the underlying webserver
+	RequestHeader set X-Forwarded-Proto "https"
+
+	# Proxify of websocket url pattern
+	ProxyPassMatch    (.*)(\/websocket)$                            ws://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1$2
```
Comment (vsellier) [inline etc/apache2/includes/instance-ws-oo.include.template]: * this should be on the
* you should add the specific ws configuration

File: etc/apache2/includes/instance-chat-standalone-cmis-oo.include.template [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+        ProxyPass /chatServer/cometd ws://localhost:${DEPLOYMENT_CHAT_SERVER_PORT}/chatServer/cometd max=200 acquire=5000 retry=5 disablereuse=on flushpackets=on
+    </IfModule>
+
+    ProxyPass /cmis http://localhost:${DEPLOYMENT_CMIS_HTTP_PORT}/cmis acquire=1000 retry=30 timeout=20
+    ProxyPassReverse /cmis http://localhost:${DEPLOYMENT_CMIS_HTTP_PORT}/cmis
+
+    # ACC-90: need to force disablereuse and flushpackets to avoid WebSocket interferences
+    # Can be remove if apache version is >= 2.4.13
+    ProxyPass               /chatServer       http://localhost:${DEPLOYMENT_CHAT_SERVER_PORT}/chatServer acquire=1000 retry=30 disablereuse=on flushpackets=on
+    ProxyPassReverse        /chatServer       htpp://localhost:${DEPLOYMENT_CHAT_SERVER_PORT}/chatServer
```
Comment (vsellier) [inline etc/apache2/includes/instance-chat-standalone-cmis-oo.include.template]: typo
```suggestion
    ProxyPassReverse        /chatServer       http://localhost:${DEPLOYMENT_CHAT_SERVER_PORT}/chatServer
```

File: etc/apache2/includes/instance-oo.include.template (hunk 3 above)
Comment (madhifallah) [reply etc/apache2/includes/instance-oo.include.template]: fixed in all templates

File: etc/apache2/includes/instance-ws-cmis-oo.include.template (hunk 5 above)
Comment (madhifallah) [reply etc/apache2/includes/instance-ws-cmis-oo.include.template]: fixed in all templates

File: etc/apache2/includes/instance-chat-standalone-cmis-oo.include.template [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    ProxyPassMatch    "^\/(v[\d]+\.[\d]+\.[\d]+[\.|-][\d]+\/.*)$"    http://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1 acquire=1000 retry=30
+    ProxyPassMatch	  "^\/OfficeWeb\/.*)$"							 http://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1 acquire=1000 retry=30
+
+    # Add header for X-Forwarded-For / X-Forwarded-Host / X-Forwarded-Server
+    ProxyAddHeaders On
+    # Add header X-Forwarded-Proto to enforce https on the underlying webserver
+    RequestHeader set X-Forwarded-Proto "https"
+
+    # Proxify of websocket url pattern
+    ProxyPassMatch    (.*)(\/websocket)$                            ws://localhost:${DEPLOYMENT_ONLYOFFICE_HTTP_PORT}/$1$2
```
[the remark this answers is not in the document]
Comment (madhifallah) [reply etc/apache2/includes/instance-chat-standalone-cmis-oo.include.template]: fixed in all templates

**Pull request merged**